home-assistant: 2021.2.3 -> 2021.3.0

mweinelt opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

https://www.home-assistant.io/blog/2021/03/03/release-20213/

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

mweinelt committed:
home-assistant: enable a lot more component tests

We can now track a list of components, that should be integrated into
our testsuite run. Their dependencies will automatically be included
into the checkInputs by reusing the dependency lookup infrastructure we
already have, which is what makes this feasible.

mweinelt committed:
python3Packages.zha-quirks: 0.0.53 -> 0.0.54

mweinelt committed:
home-assistant: rerun tests failing with RuntimeError

The flakyness in home-assistant stems from premature event loop closing,
due to race conditions throughout the test suite.

> RuntimeError: Event loop is closed

They also happen during upstreams testsuite runs, but they silently
ignore them, so I don't expect a fix for them in the near future.

Alleviate our pains by rerunning tests that fail in this particular way.

mweinelt committed:
home-assistant: 2021.2.3 -> 2021.3.0

mweinelt commented:
```
          Outdated dependencies           
┏━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┓
┃ Package             ┃ Current ┃ Wanted ┃
┡━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━┩
│ HAP-python          │ 3.1.0   │ 3.3.2  │
│ accuweather         │ 0.0.11  │ 0.1.0  │
│ adext               │ 0.3     │ 0.4.1  │
│ aioshelly           │ 0.5.4   │ 0.6.1  │
│ bellows             │ 0.21.0  │ 0.22.0 │
│ httplib2            │ 0.18.1  │ 0.19.0 │
│ pillow              │ 8.0.1   │ 8.1.1  │
│ plexapi             │ 4.3.1   │ 4.4.0  │
│ py-cpuinfo          │ 5.0.0   │ 7.0.0  │
│ pyinsteon           │ 1.0.8   │ 1.0.9  │
│ pylutron            │ 0.2.6   │ 0.2.7  │
│ pymyq               │ 3.0.3   │ 3.0.4  │
│ python-telegram-bot │ 13.0    │ 13.1   │
│ pytile              │ 5.1.1   │ 5.2.0  │
│ pyvera              │ 0.3.11  │ 0.3.13 │
│ sentry-sdk          │ 0.19.5  │ 0.20.3 │
│ sqlalchemy          │ 1.3.20  │ 1.3.23 │
│ subarulink          │ 0.3.11  │ 0.3.12 │
│ xknx                │ 0.17.0  │ 0.17.1 │
│ zigpy-znp           │ 0.3.0   │ 0.4.0  │
└─────────────────────┴─────────┴────────┘
```

fabaff commented on pkgs/servers/home-assistant/component-packages.nix:
```diff
@@ -860,7 +868,7 @@
     "trafikverket_train" = ps: with ps; [ pytrafikverket ];
     "trafikverket_weatherstation" = ps: with ps; [ pytrafikverket ];
-    "transmission" = ps: with ps; [ transmissionrpc ];
+    "transmission" = ps: with ps; [ ]; # missing inputs: transmissionrpc
```
Looks like that the package was [updated](https://github.com/NixOS/nixpkgs/tree/nixos-unstable/pkgs/development/python-modules/transmission-rpc) (from 0.11 to 3.2.2) and renamed in #112237.

 I think that without some feedback of transmission user it's hard to tell with the integration will work with 3.2.2.

mweinelt replied:
Worse.

```
{
  "domain": "transmission",
  "name": "Transmission",
  "config_flow": true,
  "documentation": "https://www.home-assistant.io/integrations/transmission",
  "requirements": ["transmissionrpc==0.11"],
  "codeowners": ["@engrbm87", "@JPHutchins"]
}
```

They are two different packages.

https://pypi.org/project/transmissionrpc/
https://pypi.org/project/transmission-rpc/

Mic92 replied:
I actually had to downgrade 3.2.2 back to 0.11 because it no longer worked.

fabaff replied:
It needs an entry in `PKG_PREFERENCES` of `parse-requirements.py`.

I guess that Martin will run `parse-requirements.py` before finishing this PR anyways and with the new entry in the dict we will end-up a `component-packages.nix` that contains `transmissionrpc` again.

mweinelt replied:
I tried exactly that yesterday,

```diff
diff --git a/pkgs/servers/home-assistant/parse-requirements.py b/pkgs/servers/home-assistant/parse-requirements.py
index eeb117a984e..18577e68f36 100755
--- a/pkgs/servers/home-assistant/parse-requirements.py
+++ b/pkgs/servers/home-assistant/parse-requirements.py
@@ -42,6 +42,7 @@ PKG_PREFERENCES = {
     "tensorflow-bin_2": "tensorflow",
     "tensorflowWithoutCuda": "tensorflow",
     "tensorflow-build_2": "tensorflow",
+    "transmissionrpc": "transmission-rpc",
     "whois": "python-whois",
 }
```

but that went nowhere.

fabaff replied:
In #112237 it looked like that `transmission-rpc` is a drop-in replacement for `transmissionrpc`. But the it's a fork which introduces a new module. 

https://github.com/NixOS/nixpkgs/pull/115186 will re-add `transmissionrpc`.

mweinelt replied:
Rebased, thank you!

fabaff commented:
Nice job on the tests by the way.

fabaff commented:
>           Outdated dependencies           

I already have some on my todo list. Some updates will require `aiohttp-3.7.4`.

mweinelt commented:
> > ```
> >       Outdated dependencies           
> > ```
> 
> I already have some on my todo list. Some updates will require `aiohttp-3.7.4`.

Already in staging.

https://github.com/NixOS/nixpkgs/pull/114556

mweinelt committed:
python3Packages.zigpy-znp: 0.3.0 -> 0.4.0

mweinelt committed:
python3Pacakges.pyvera: 0.3.11 -> 0.3.13

fabaff commented:
Oh, I was not playing attention to this PR. You already updated the some of dependencies.

I will close #115020.

mweinelt commented:
Already merged and rebased this on top.

mweinelt committed:
python3Packages.bellows: 0.21.0 -> 0.22.0

mweinelt committed:
python3Packages.alarmdecoder: 1.13.9 -> 1.13.10

mweinelt committed:
python3Packaegs.adext: 0.3 -> 0.4.1

mweinelt committed:
python3Packages.accuweather: 0.0.11 -> 0.1.0

mweinelt committed:
python3Packages.plexapi: 4.3.1 -> 4.4.0

mweinelt commented:
Down to

```
          Outdated dependencies           
┏━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┓
┃ Package             ┃ Current ┃ Wanted ┃
┡━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━┩
│ HAP-python          │ 3.1.0   │ 3.3.2  │
│ httplib2            │ 0.18.1  │ 0.19.0 │
│ pillow              │ 8.0.1   │ 8.1.1  │
│ py-cpuinfo          │ 5.0.0   │ 7.0.0  │
│ python-telegram-bot │ 13.0    │ 13.1   │
│ pytile              │ 5.1.1   │ 5.2.0  │
│ sentry-sdk          │ 0.19.5  │ 0.20.3 │
│ sqlalchemy          │ 1.3.20  │ 1.3.23 │
└─────────────────────┴─────────┴────────┘
```

Already in staging:
- httplib2
- pillow
- sqlalchemy
- py-cpuinfo

dotlambda commented:
We might not want all users of the Home Assistant package to have to test so many components. Maybe we can create a new package called `home-assistantWithTests` which sets `doCheck = true` and tests as much as possible -- so that maintainers notice when something breaks.

mweinelt commented:
Or we document (e.g. in the wiki), how users can disable tests.

```nix
{ pkgs, ... }:

{
  services.home-assistant = {
    enable = true;
    package = pkgs.home-assistant.overrideAttrs (old: {
      doCheck = false;
      doInstallCheck = false;
    });
  };
}
```

dotlambda commented:
I would use `overridePythonAttrs` and only set `doCheck = false`.

dotlambda commented:
I guess it's enough to document this as an example for the `package` option and by mentioning in the option's description that otherwise building might take a long time.
We should also the default `package` one that doesn't run the tests.

mweinelt commented:
I tried the following, but it results in the tests not finding aiohttp_cors.

```patch
diff --git a/nixos/modules/services/misc/home-assistant.nix b/nixos/modules/services/misc/home-assistant.nix
index 1f2e13f3732..7d0a031be72 100644
--- a/nixos/modules/services/misc/home-assistant.nix
+++ b/nixos/modules/services/misc/home-assistant.nix
@@ -183,8 +183,14 @@ in {
     };
 
     package = mkOption {
-      default = pkgs.home-assistant;
-      defaultText = "pkgs.home-assistant";
+      default = pkgs.home-assistant.overridePythonAttrs (oldAttrs: {
+        doCheck = false;
+      });
+      defaultText = literalExample ''
+        pkgs.home-assistant.overridePythonAttrs (oldAttrs: {
+          doCheck = false;
+        }
+      '';
       type = types.package;
       example = literalExample ''
         pkgs.home-assistant.override {
```

```
hass # [    5.818279] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain logger took 0.0 seconds
hass # [    5.819942] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up http
hass # [    5.825566] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Error during setup of component http
hass # [    5.826350] hass[722]: Traceback (most recent call last):
hass # [    5.826836] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/setup.py", line 213, in _async_setup_component
hass # [    5.828039] hass[722]:     result = await task
hass # [    5.828470] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/__init__.py", line 210, in async_setup
hass # [    5.830821] hass[722]:     server = HomeAssistantHTTP(
hass # [    5.831271] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/__init__.py", line 316, in __init__
hass # [    5.832405] hass[722]:     setup_cors(app, cors_origins)
hass # [    5.833560] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/cors.py", line 26, in setup_cors
hass # [    5.834864] hass[722]:     import aiohttp_cors
hass # [    5.835553] hass[722]: ModuleNotFoundError: No module named 'aiohttp_cors'
hass # [    5.836764] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of system_log. Setup failed for dependencies: http
hass # [    5.838257] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for system_log: Could not set up all dependencies.
hass # [    5.839609] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 1: {'image', 'lovelace', 'device_automation', 'auth', 'person', 'search', 'system_log', 'onboarding', 'frontend', 'websocket_api', 'api', 'config', 'http'}
hass # [    5.867062] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up lovelace
hass # [    5.874013] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up device_automation
hass # [    5.874873] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain device_automation took 0.0 seconds
hass # [    5.876540] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of image. Setup failed for dependencies: http
hass # [    5.877624] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for image: Could not set up all dependencies.
hass # [    5.878799] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of auth. Setup failed for dependencies: http
hass # [    5.879682] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for auth: Could not set up all dependencies.
hass # [    5.881515] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of websocket_api. Setup failed for dependencies: http
hass # [    5.883960] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for websocket_api: Could not set up all dependencies.
hass # [    5.885806] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of api. Setup failed for dependencies: http
hass # [    5.886679] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for api: Could not set up all dependencies.
hass # [    5.888477] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of config. Setup failed for dependencies: http
hass # [    5.890927] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for config: Could not set up all dependencies.
hass # [    5.892819] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of person. Setup failed for dependencies: image
hass # [    5.894459] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for person: Could not set up all dependencies.
hass # [    5.895397] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of search. Setup failed for dependencies: websocket_api
hass # [    5.896604] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for search: Could not set up all dependencies.
hass # [    5.897987] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain lovelace took 0.0 seconds
hass # [    5.899008] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of onboarding. Setup failed for dependencies: auth, http, person
hass # [    5.900292] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for onboarding: Could not set up all dependencies.
hass # [    5.901780] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of frontend. Setup failed for dependencies: api, auth, config, http, onboarding, search, system_log, websocket_api
hass # [    5.903498] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for frontend: Could not set up all dependencies.
hass # [    5.904973] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 2: {'binary_sensor', 'mqtt'}
hass # [    5.957993] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up binary_sensor
hass # [    5.958908] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain binary_sensor took 0.0 seconds
hass # [    5.960115] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of mqtt. Setup failed for dependencies: http
hass # [    5.961802] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for mqtt: Could not set up all dependencies.
hass # [    5.963800] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of mqtt. Setup failed for dependencies: http
hass # [    5.964682] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform mqtt.binary_sensor: Could not set up all dependencies.
hass # [    5.966991] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Home Assistant initialized in 0.20s
hass # [    5.967995] hass[722]: 2021-03-03 23:58:00 WARNING (MainThread) [homeassistant.bootstrap] Detected that frontend did not load. Activating safe mode
hass # [    5.970798] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Starting in safe mode
hass # [    5.971548] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded homeassistant from homeassistant.components.homeassistant
hass # [    5.973184] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded persistent_notification from homeassistant.components.persistent_notification
hass # [    5.974796] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up homeassistant
hass # [    5.976160] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain homeassistant took 0.0 seconds
hass # [    5.978032] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up persistent_notification
hass # [    5.979673] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain persistent_notification took 0.0 seconds
hass # [    5.982186] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded http from homeassistant.components.http
hass # [    5.983808] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded safe_mode from homeassistant.components.safe_mode
hass # [    5.985548] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_3) [homeassistant.loader] Loaded frontend from homeassistant.components.frontend
hass # [    5.987764] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded api from homeassistant.components.api
hass # [    5.988971] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_2) [homeassistant.loader] Loaded auth from homeassistant.components.auth
hass # [    5.990722] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded config from homeassistant.components.config
hass # [    5.992177] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_3) [homeassistant.loader] Loaded device_automation from homeassistant.components.device_automation
hass # [    5.993794] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded lovelace from homeassistant.components.lovelace
hass # [    5.995001] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_2) [homeassistant.loader] Loaded onboarding from homeassistant.components.onboarding
hass # [    6.000162] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded person from homeassistant.components.person
hass # [    6.000976] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_3) [homeassistant.loader] Loaded image from homeassistant.components.image
hass # [    6.001750] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded search from homeassistant.components.search
hass # [    6.003573] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_2) [homeassistant.loader] Loaded websocket_api from homeassistant.components.websocket_api
hass # [    6.005409] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded system_log from homeassistant.components.system_log
hass # [    6.007899] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_3) [homeassistant.loader] Loaded cloud from homeassistant.components.cloud
hass # [    6.008720] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded webhook from homeassistant.components.webhook
hass # [    6.010566] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_2) [homeassistant.loader] Loaded alexa from homeassistant.components.alexa
hass # [    6.012345] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Domains to be set up: {'image', 'cloud', 'device_automation', 'auth', 'lovelace', 'person', 'search', 'webhook', 'api', 'config', 'alexa', 'onboarding', 'frontend', 'websocket_api', 'system_log', 'safe_mode', 'persistent_notification', 'http'}
hass # [    6.015901] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up logging: {'system_log'}
hass # [    6.016635] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up http
hass # [    6.018337] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Error during setup of component http
hass # [    6.020910] hass[722]: Traceback (most recent call last):
hass # [    6.021378] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/setup.py", line 213, in _async_setup_component
hass # [    6.023949] hass[722]:     result = await task
hass # [    6.024384] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/__init__.py", line 210, in async_setup
hass # [    6.026205] hass[722]:     server = HomeAssistantHTTP(
hass # [    6.026791] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/__init__.py", line 316, in __init__
hass # [    6.028530] hass[722]:     setup_cors(app, cors_origins)
hass # [    6.029462] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/cors.py", line 26, in setup_cors
hass # [    6.031272] hass[722]:     import aiohttp_cors
hass # [    6.031698] hass[722]: ModuleNotFoundError: No module named 'aiohttp_cors'
hass # [    6.032261] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of system_log. Setup failed for dependencies: http
hass # [    6.033141] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for system_log: Could not set up all dependencies.
hass # [    6.033976] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 1: {'cloud', 'image', 'device_automation', 'auth', 'lovelace', 'webhook', 'person', 'search', 'system_log', 'alexa', 'onboarding', 'frontend', 'websocket_api', 'api', 'config', 'http'}
hass # [    6.035361] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up device_automation
hass # [    6.036074] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain device_automation took 0.0 seconds
hass # [    6.036845] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up lovelace
hass # [    6.037484] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain lovelace took 0.0 seconds
hass # [    6.038243] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of image. Setup failed for dependencies: http
hass # [    6.039171] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for image: Could not set up all dependencies.
hass # [    6.039995] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of auth. Setup failed for dependencies: http
hass # [    6.040861] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for auth: Could not set up all dependencies.
hass # [    6.041664] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of webhook. Setup failed for dependencies: http
hass # [    6.042631] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for webhook: Could not set up all dependencies.
hass # [    6.043465] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of alexa. Setup failed for dependencies: http
hass # [    6.044353] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for alexa: Could not set up all dependencies.
hass # [    6.045184] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of websocket_api. Setup failed for dependencies: http
hass # [    6.046142] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for websocket_api: Could not set up all dependencies.
hass # [    6.047063] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of api. Setup failed for dependencies: http
hass # [    6.047959] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for api: Could not set up all dependencies.
hass # [    6.048780] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of config. Setup failed for dependencies: http
hass # [    6.049675] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for config: Could not set up all dependencies.
hass # [    6.050509] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of person. Setup failed for dependencies: image
hass # [    6.051456] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for person: Could not set up all dependencies.
hass # [    6.052311] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of cloud. Setup failed for dependencies: http, webhook, alexa
hass # [    6.053262] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for cloud: Could not set up all dependencies.
hass # [    6.054071] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of search. Setup failed for dependencies: websocket_api
hass # [    6.055005] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for search: Could not set up all dependencies.
hass # [    6.055954] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of onboarding. Setup failed for dependencies: auth, http, person
hass # [    6.056956] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for onboarding: Could not set up all dependencies.
hass # [    6.057795] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of frontend. Setup failed for dependencies: api, auth, config, http, onboarding, search, system_log, websocket_api
hass # [    6.058952] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for frontend: Could not set up all dependencies.
hass # [    6.059784] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 2: {'safe_mode', 'persistent_notification'}
hass # [    6.060690] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of safe_mode. Setup failed for dependencies: frontend, cloud
hass # [    6.061661] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for safe_mode: Could not set up all dependencies.
hass # [    6.062478] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Home Assistant initialized in 0.06s
hass # [    6.063211] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.core] Starting Home Assistant
hass # [    6.063888] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.core] Timer:starting
```

mweinelt commented:
@GrahamcOfBorg eval

SuperSandro2000 commented on pkgs/development/python-modules/accuweather/default.nix:
```diff
@@ -1,22 +1,43 @@
 
+  postPatch = ''
+    # we don't have pytest-error-for-skips packaged
+    sed -i "s/--error-for-skips//" pytest.ini
```
```suggestion
    substituteInPlace pytest.ini --replace '--error-for-skips' ""
```

SuperSandro2000 commented on pkgs/development/python-modules/adext/default.nix:
```diff
@@ -1,24 +1,26 @@
 { lib
 , buildPythonPackage
 , fetchPypi
+, setuptools_scm
```
```suggestion
, setuptools-scm
```

SuperSandro2000 commented on pkgs/development/python-modules/adext/default.nix:
```diff
@@ -1,24 +1,26 @@
-      --replace "alarmdecoder==1.13.2" "alarmdecoder>=1.13.2"
-  '';
+  nativeBuildInputs = [
+    setuptools_scm
```
```suggestion
    setuptools-scm
```

SuperSandro2000 commented on pkgs/development/python-modules/pyvera/default.nix:
```diff
@@ -13,24 +13,16 @@
   };
 
-  patches = [
-    (fetchpatch {
```
Please remove fetchpatch from inputs.

SuperSandro2000 commented on pkgs/servers/home-assistant/component-packages.nix:
```diff
@@ -120,13 +121,13 @@
     "cmus" = ps: with ps; [ ]; # missing inputs: pycmus
     "co2signal" = ps: with ps; [ ]; # missing inputs: co2signal
     "coinbase" = ps: with ps; [ ]; # missing inputs: coinbase
-    "coinmarketcap" = ps: with ps; [ coinmarketcap ];
```
Do we want to remove this python package from nixpkgs?

mweinelt replied:
Why? Because home-assistant is the only visible consumer?

SuperSandro2000 replied:
Yes and it is also maintainerless.

mweinelt replied:
https://github.com/home-assistant/core/pull/46615

Mic92 replied:
See https://github.com/NixOS/nixpkgs/pull/115050

dotlambda commented:
> I tried the following, but it results in the tests not finding aiohttp_cors.

No idea why it doesn't work.
Another suggestion: add an argument `doCheck` to the expression, so that we can just use `override`. You pick what you prefer.

mweinelt commented:
Two open test issues remain:

```
___________________ test_fetching_without_verify_ssl[pyloop] ___________________
[gw3] linux -- Python 3.8.8 /nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/bin/python3.8

aioclient_mock = <tests.test_util.aiohttp.AiohttpClientMocker object at 0x7fff6f657460>
hass = <homeassistant.core.HomeAssistant object at 0x7fff6f71ae20>
hass_client = <function hass_client.<locals>.auth_client at 0x7fff6f667ca0>

    async def test_fetching_without_verify_ssl(aioclient_mock, hass, hass_client):
        """Test that it fetches the given url when ssl verify is off."""
        aioclient_mock.get("https://example.com", text="hello world")

        await async_setup_component(
            hass,
            "camera",
            {
                "camera": {
                    "name": "config_test",
                    "platform": "generic",
                    "still_image_url": "https://example.com",
                    "username": "user",
                    "password": "pass",
                    "verify_ssl": "false",
                }
            },
        )
        await hass.async_block_till_done()

        client = await hass_client()

        resp = await client.get("/api/camera_proxy/camera.config_test")

>       assert resp.status == 200
E       AssertionError: assert 500 == 200
E        +  where 500 = <ClientResponse(http://127.0.0.1:43291/api/camera_proxy/camera.config_test) [500 Internal Server Error]>\n<CIMultiDictP...harset=utf-8', 'Content-Length': '26', 'Date': 'Thu, 04 Mar 2021 08:06:56 GMT', 'Server': 'Python/3.8 aiohttp/3.7.3')>\n.status

tests/components/generic/test_camera.py:76: AssertionError
```

```
__________________ test_fetching_url_with_verify_ssl[pyloop] ___________________
[gw3] linux -- Python 3.8.8 /nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/bin/python3.8

aioclient_mock = <tests.test_util.aiohttp.AiohttpClientMocker object at 0x7fff6f62dc10>
hass = <homeassistant.core.HomeAssistant object at 0x7fff6f6c2c40>
hass_client = <function hass_client.<locals>.auth_client at 0x7fff6f7dc040>

    async def test_fetching_url_with_verify_ssl(aioclient_mock, hass, hass_client):
        """Test that it fetches the given url when ssl verify is explicitly on."""
        aioclient_mock.get("https://example.com", text="hello world")

        await async_setup_component(
            hass,
            "camera",
            {
                "camera": {
                    "name": "config_test",
                    "platform": "generic",
                    "still_image_url": "https://example.com",
                    "username": "user",
                    "password": "pass",
                    "verify_ssl": "true",
                }
            },
        )
        await hass.async_block_till_done()

        client = await hass_client()

        resp = await client.get("/api/camera_proxy/camera.config_test")

>       assert resp.status == 200
E       AssertionError: assert 500 == 200
E        +  where 500 = <ClientResponse(http://127.0.0.1:38267/api/camera_proxy/camera.config_test) [500 Internal Server Error]>\n<CIMultiDictP...harset=utf-8', 'Content-Length': '26', 'Date': 'Thu, 04 Mar 2021 08:06:56 GMT', 'Server': 'Python/3.8 aiohttp/3.7.3')>\n.status

tests/components/generic/test_camera.py:103: AssertionError
```

fabaff commented:
Result of `nixpkgs-review pr 115016` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python37Packages.accuweather</li>
    <li>python37Packages.adext</li>
    <li>python37Packages.alarmdecoder</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.plexapi</li>
    <li>python37Packages.pyvera</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.accuweather</li>
    <li>python38Packages.adext</li>
    <li>python38Packages.alarmdecoder</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.plexapi</li>
    <li>python38Packages.pyvera</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.accuweather</li>
    <li>python39Packages.adext</li>
    <li>python39Packages.alarmdecoder</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.pyvera</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy-znp</li>
  </ul>
</details>

mweinelt commented:
> building of '/nix/store/a8hlrwq6vy08asviqib3wh78lzpmiryn-homeassistant-2021.3.0.drv!out' from .drv file timed out after 3600 seconds

The aarch64 ofborg builder seems to timeout near the end of the tests. I have a raspberrypi4 here, but could someone else give the nixos test on aarch64 a try?

fabaff commented:
Just as reference. No action needed and not need to be part of this PR. I simply want to avoid duplicate work like it happen yesterday.

New dependencies according to `component-packages.nix`:

- aiolyric: #115125
- pyplaato: #115124
- pyrituals: #115119
- mullvad-api: #115113

Package updates (https://github.com/NixOS/nixpkgs/pull/115016#issuecomment-790086395)

- HAP-python: #115087

mweinelt commented:
> > building of '/nix/store/a8hlrwq6vy08asviqib3wh78lzpmiryn-homeassistant-2021.3.0.drv!out' from .drv file timed out after 3600 seconds
> 
> The aarch64 ofborg builder seems to timeout near the end of the tests. I have a raspberrypi4 here, but could someone else give the nixos test on aarch64 a try, too?

There is a problem I can reproduce on aarch64, where a test times out, maybe multiple.

> tests/components/zha/test_discover.py::test_group_probe_cleanup_called[pyloop]

mvnetbiz reviewed (commented):
Frontend is broken for me, I can't log in on any computers. A few things are still working from the android app though.

mvnetbiz reviewed (approved):
Nevermind, LGTM, must have been some weird browser cache thing or something, even though I tried multiple browsers on my laptop and phone, and a fresh firefox profile :/

fabaff commented:
> There is a problem I can reproduce on aarch64, where a test times out, maybe multiple.

I'm stuck on a Raspberry Pi at 99 %. I guess that the timeout is coming as well.

```bash
builder for '/nix/store/5952fjjxy1hlgij5rmdrcdkkx4biwg17-python3.7-zigpy-znp-0.4.0.drv' failed with exit code 1; last 10 log lines:
  
  final      = True
  input      = b''
  self       = <encodings.utf_8.IncrementalDecoder object at 0x7ff5f367d0>
  
  /nix/store/dxmx8nx86d45g3wnp6xvrs8l8wsclc1r-python3-3.7.10/lib/python3.7/codecs.py:319: Failed
  =========================== short test summary info ============================
  FAILED tests/application/test_requests.py::test_request_concurrency[FormedLaunchpadCC26X2R1]
  ERROR tests/tools/test_nvram.py::test_nvram_write[ResetZStack3CC2531] - Faile...
  ============== 1 failed, 334 passed, 1 error in 260.87s (0:04:20) ==============
[...]
[1/0/2 built (1 failed), 0.0 MiB DL] building homeassistant-2021.3.0 (pytestCheckPhase): ........................................................................ [ 99%]
```

mweinelt commented:
My current progress is this:

```
diff --git a/pkgs/servers/home-assistant/default.nix b/pkgs/servers/home-assistant/default.nix
index 1e5ea0f7532..b6260e238e6 100644
--- a/pkgs/servers/home-assistant/default.nix
+++ b/pkgs/servers/home-assistant/default.nix
@@ -132,6 +132,7 @@ in with py.pkgs; buildPythonApplication rec {
     asynctest
     pytest-aiohttp
     pytest-rerunfailures
+    pytest-timeout
     pytest-xdist
     pytestCheckHook
     requests-mock
@@ -287,6 +288,8 @@ in with py.pkgs; buildPythonApplication rec {
   ];
 
   pytestFlagsArray = [
+    "-vvv"
+    "--timeout 60"
     # limit amout of runners to reduce race conditions
     "-n auto"
     # retry racy tests that end in "RuntimeError: Event loop is closed"
@@ -304,6 +307,11 @@ in with py.pkgs; buildPythonApplication rec {
     "tests/components"
     # pyotp since v2.4.0 complains about the short mock keys, hass pins v2.3.0
     "tests/auth/mfa_modules/test_notify.py"
+  ] ++ lib.optionals (stdenv.isAarch64) [
+    # tests get stuck on aarch64
+    "tests/components/zha/test_discover.py"
+    "tests/components/zwave/test_init.py"
+    "tests/components/zwave/test_climate.py"
   ];
 
   disabledTests = [

```

Sadly the timeout never triggers, it just gets stuck at 99%.

mweinelt commented:
> ```shell
> builder for '/nix/store/5952fjjxy1hlgij5rmdrcdkkx4biwg17-python3.7-zigpy-znp-0.4.0.drv' failed with exit code 1; last 10 log lines:
>   
>   final      = True
>   input      = b''
>   self       = <encodings.utf_8.IncrementalDecoder object at 0x7ff5f367d0>
>   
>   /nix/store/dxmx8nx86d45g3wnp6xvrs8l8wsclc1r-python3-3.7.10/lib/python3.7/codecs.py:319: Failed
>   =========================== short test summary info ============================
>   FAILED tests/application/test_requests.py::test_request_concurrency[FormedLaunchpadCC26X2R1]
>   ERROR tests/tools/test_nvram.py::test_nvram_write[ResetZStack3CC2531] - Faile...
>   ============== 1 failed, 334 passed, 1 error in 260.87s (0:04:20) ==============
> [...]
> [1/0/2 built (1 failed), 0.0 MiB DL] building homeassistant-2021.3.0 (pytestCheckPhase): ........................................................................ [ 99%]
> ```

Can't reproduce on aarch64, nor on x86_64.

fabaff commented:
> Can't reproduce on aarch64, nor on x86_64.

First time I saw it was on the Raspberry today. x86_64 is fine for me.

fabaff commented:
Result of `nixpkgs-review pr 115016` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python37Packages.accuweather</li>
    <li>python37Packages.adext</li>
    <li>python37Packages.alarmdecoder</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.plexapi</li>
    <li>python37Packages.pyvera</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.accuweather</li>
    <li>python38Packages.adext</li>
    <li>python38Packages.alarmdecoder</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.plexapi</li>
    <li>python38Packages.pyvera</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.accuweather</li>
    <li>python39Packages.adext</li>
    <li>python39Packages.alarmdecoder</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.pyvera</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy-znp</li>
  </ul>
</details>

mweinelt committed:
nixos/home-assistant: disable tests on the package by default

We are running over 6000 tests by now and they take around 5 minutes
on faster machines and tests alot of components that endusers will not
actually be using. It is sufficient if we run them on package upgrades
and in the passthrough test.

mweinelt committed:
home-assistant: 2021.3.0 -> 2021.3.2

mweinelt commented:
Disabled tests in the modules default package.

Meanwhile:
- https://github.com/home-assistant/core/releases/tag/2021.3.1
- https://github.com/home-assistant/core/releases/tag/2021.3.2

Still debugging aarch64 test issues.

mweinelt commented:
> ========== 6904 passed, 5 skipped, 3251 warnings in 340.45s (0:05:40) ==========

The `esphome` component seriously needs to get their warnings in order. Several instances like these:

```
../../nix/store/nzcla6pyq2ngiacm6vzaj6m7siblhvg4-python3.8-aioesphomeapi-2.6.5/lib/python3.8/site-packages/aioesphomeapi/api_options_pb2.py:19: 32 warnings
  /nix/store/nzcla6pyq2ngiacm6vzaj6m7siblhvg4-python3.8-aioesphomeapi-2.6.5/lib/python3.8/site-packages/aioesphomeapi/api_options_pb2.py:19: DeprecationWarning: Call to deprecated create function FileDescriptor(). Note: Create unlinked descriptors is going to go away. Please use get/find descriptors from generated code or query the descriptor_pool.
    DESCRIPTOR = _descriptor.FileDescriptor(
```

mweinelt committed:
home-assistant: disable tests getting stuck on aarch64

mweinelt commented:
I've identified the two tests that get stuck on aarch64, I guess we should report them upstream and ask them to test aarch64 as well.

mweinelt commented:
https://github.com/home-assistant/core/issues/47527
